### PR TITLE
Skip invalid clusters

### PR DESCRIPTION
**Pull request by ikitiki**
— Ignore clusters with wrong specification
— catch quantity parsing errors

### CODE REVIEW

File: pkg/controller/postgresql.go [hunk 1]
```diff
@@ -37,16 +37,22 @@ func (c *Controller) clusterListFunc(options api.ListOptions) (runtime.Object, e
...
+			continue
+		}
 		c.queueClusterEvent(nil, pg, spec.EventSync)
 
 		c.logger.Debugf("Sync of the '%s' cluster has been queued", util.NameFromMeta(pg.Metadata))
+		activeClustersCnt++
 	}
-	if len(objList) > 0 {
+	if activeClustersCnt > 0 {
 		c.logger.Infof("There are %d clusters currently running", len(objList))
```
Comment (alexeyklyukin) [inline pkg/controller/postgresql.go]: Perhaps should the number of clusters that has errors. Would be useful when we have a background cleaner.

File: pkg/controller/postgresql.go [hunk 2]
```diff
@@ -168,17 +174,30 @@ func (c *Controller) processClusterEventsQueue(idx int) {
...
 	)
 
 	if old != nil {
 		uid = old.Metadata.GetUID()
 		clusterName = util.NameFromMeta(old.Metadata)
+		if eventType == spec.EventUpdate && new.Error == nil && old != nil {
+			eventType = spec.EventAdd
+			clusterError = new.Error
+		}
+		clusterError = old.Error
```
Comment (alexeyklyukin) [inline pkg/controller/postgresql.go]: should be in the else block.

File: pkg/controller/postgresql.go [hunk 3]
```diff
@@ -168,17 +174,30 @@ func (c *Controller) processClusterEventsQueue(idx int) {
...
 	} else {
 		uid = new.Metadata.GetUID()
 		clusterName = util.NameFromMeta(new.Metadata)
+		clusterError = new.Error
 	}
+
+	if clusterError != nil {
+		c.logger.Debugf("Skipping %s event for invalid cluster %s (reason: %s)", eventType, clusterName, clusterError)
```
Comment (alexeyklyukin) [inline pkg/controller/postgresql.go]: Why would we do it for the spec.EventDelete?

**Comment by alexeyklyukin** [comment]
👍

**Comment by ikitiki** [comment]
👍

**Pull request merged**